src/app/vet/(logged-in)/schedule/_lib/use-owner-contact-details.ts

```diff
@@ -0,0 +1,34 @@
+import { OwnerContactDetails } from "@/lib/vet/vet-models";
+import useSWR from "swr";
+
+async function fetchOwnerContactDetails(
+  dogIdKey: string,
+): Promise<OwnerContactDetails | undefined> {
+  if (dogIdKey === "") {
+    return undefined;
+  }
+  const url = `/api/vet/dog-owners/${dogIdKey}`;
+  const res = await fetch(url);
```

Comment:
Eh, should we check the status code? The route possibly returns 200, 401, 403, 404, and 500 (Ref1).

Ref1: https://github.com/barkbank/barkbank-pawtal/blob/fc776df31b1dd4914e93dd065c6d49a247723bc0/src/app/api/vet/dog-owners/%5BdogId%5D/route.ts

Reply:
Checks added in defc6bb